### PR TITLE
558 verify distance methods for LIME timeseries

### PR DESCRIPTION
**Pull request by geek-yang**
In this PR we verify all the calculation methods for distance, incl:

- DTW
- cosine
- Euclidean

These distance calculations are needed for LIME. Changes related to the first instance issue #578 has been adapted to the distance methods in this PR as well.

### CODE REVIEW

**Comment by geek-yang** [comment]
The failed tests are related to #587. It can be ignored when reviewing this PR.

**Comment by elboyran** [comment]
@cwmeijer seems important PR. Will you, please, review it?

File: tests/methods/test_lime_timeseries.py [hunk 1]
```diff
@@ -26,30 +26,55 @@ def test_lime_timeseries_correct_output_shape(self):
...
+        distance_euclidean = explainer._calculate_distance(
+            masked_data, distance_method="euclidean"
+        )
+
+        # Check that the calculated and expected results are equal
+        np.testing.assert_array_almost_equal(
+            [expected_cosine_distance, expected_euclidean_distance],
+            [distance_cosine, distance_euclidean],
+        )
+
```
Comment (stefsmeets) [inline tests/methods/test_lime_timeseries.py:66]: These are essentially two unit tests in one. I would suggest to split this up into two tests, this makes it easier to see what is going on when a test fails.

```suggestion
    def test_euclidean_distance(self):
        """Test the calculation of cosine and euclidean distance."""
        masked_data = np.array([[1, 2, 3, 4, 5], [0, 0, 3, 4, 5]])
        expected_distance = np.array([0.0, 2.236068])

        explainer = LIMETimeseries()
        distance = explainer._calculate_distance(masked_data,
                                                 distance_method="euclidean")
        np.testing.assert_array_almost_equal(distance, expected_distance)

    def test_cosine_distance(self):
        """Test the calculation of cosine and euclidean distance."""
        masked_data = np.array([[1, 2, 3, 4, 5], [0, 0, 3, 4, 5]])
        expected_distance = np.array([0.0, 4.653741])

        explainer = LIMETimeseries()
        distance = explainer._calculate_distance(masked_data,
                                                 distance_method="cosine")
        np.testing.assert_array_almost_equal(distance, expected_distance)

```

**Review by stefsmeets — APPROVED** [review]
Nice work, looks good to me! Just a small suggestion to improve the tests by splitting it into two separate tests.

File: tests/methods/test_lime_timeseries.py (hunk 1 above)
Comment (geek-yang) [reply tests/methods/test_lime_timeseries.py]: Thanks for your review @stefsmeets. That makes sense to me. I will split it into two tests.

**Pull request merged**